ibis/pyspark/tests/test_timecontext.py

```diff
@@ -9,7 +14,408 @@
+    time       value  count
+    2020-01-03   c      2
+    2020-01-04   d      2
+    trailing count for 2 days will be:
+    time       value  count
+    2020-01-03   c      3
+    2020-01-04   d      3
+    with context = (2020-01-01, 2002-01-02) count for 1 day forward looking
+    window will be:
+    time       value  count
+    2020-01-01   a      2
+    2020-01-02   b      2
+    """
+    table = client.table('time_indexed_table')
+    context = (
+        pd.Timestamp('20170102 07:00:00', tz='UTC'),
+        pd.Timestamp('20170103', tz='UTC'),
+    )
+    result_pd = table.mutate(
+        count=table['value'].count().over(ibis_window[0])
+    ).execute(timecontext=context)
+    spark_table = table.compile()
+    spark_window = (
+        Window.partitionBy('key')
+        .orderBy(F.col('time').cast('long'))
+        .rangeBetween(*spark_range)
+    )
+    expected = spark_table.withColumn(
+        'count', F.count(spark_table['value']).over(spark_window),
+    ).toPandas()
+    expected = expected[
+        expected.time.between(*(t.tz_convert(None) for t in context))
+    ].reset_index(drop=True)
+    tm.assert_frame_equal(result_pd, expected)
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [([('cumulative', 1)], (Window.unboundedPreceding, 0))],
+    indirect=['ibis_window'],
+)
+def test_cumulative_window(client, ibis_window, spark_range):
+    """ Test context adjustment for cumulative window
+
+    For cumulative window, by defination we should look back infinately.
+    When data is trimmed by time context, we define the limit of looking
+    back is the start time of given time context. Thus for a table of
+    time       value
+    2020-01-01   a
+    2020-01-02   b
+    2020-01-03   c
+    2020-01-04   d
+    with context = (2020-01-02, 2002-01-03) cumulative count will be:
+    time       value  count
+    2020-01-02   b      1
+    2020-01-03   c      2
+    """
+    table = client.table('time_indexed_table')
+    context = (
+        pd.Timestamp('20170102 07:00:00', tz='UTC'),
+        pd.Timestamp('20170105', tz='UTC'),
+    )
+    result_pd = table.mutate(
+        count_cum=table['value'].count().over(ibis_window[0])
+    ).execute(timecontext=context)
+
+    spark_table = table.compile(timecontext=context)
+    spark_window = (
+        Window.partitionBy('key')
+        .orderBy(F.col('time').cast('long'))
+        .rangeBetween(*spark_range)
+    )
+    expected = spark_table.withColumn(
+        'count_cum', F.count(spark_table['value']).over(spark_window),
+    ).toPandas()
+    expected = expected[
+        expected.time.between(*(t.tz_convert(None) for t in context))
+    ].reset_index(drop=True)
+    tm.assert_frame_equal(result_pd, expected)
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [([('trailing', 1), ('trailing', 2)], [(-3600, 0), (-7200, 0)])],
+    indirect=['ibis_window'],
+)
+def test_multiple_trailing_window(client, ibis_window, spark_range):
```

Comment:
Can you add a test that has dependent trailing windows? e.g.

```
table = table.mutate(new_col=table['v'].mean().over(win1))
table = table.mutate(new_col2=table['new_col'].mean().over(win2))
```

Reply:
Sure, I added one